### PR TITLE
Updating Schema reading procedures and refactoring

### PR DESCRIPTION
**Pull request by Hsankesara**
- [Updated data reading procedure](https://github.com/RADAR-base/radarpipeline/commit/faef72958028775a03a1c292bf90764dda8431ea)
- [refactor to make all modules more spark dependent](https://github.com/RADAR-base/radarpipeline/commit/fba2d7552e0cb5c447f33128c10f4fbd71a73643)
- [Added and updated tests + upgraded spark version to 3.5.0](https://github.com/RADAR-base/radarpipeline/pull/78/commits/25dd2374671dea801ceb145a216d9015640f241e)

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ project:
     version: mock_version
 
 input:
-    data_type: local # couldbe mock, local, sftp, s3
+    data_type: mock # couldbe mock, local, sftp, s3
```
Comment (afolarin) [inline config.yaml]: would this be better as `data_source` or `source_type` data type is more specific to the data, this I think relates more to the source of the data

File: radarpipeline/io/downloader.py
Comment (afolarin) [inline radarpipeline/io/downloader.py:1]: While SFTP (though rsync might be useful for restart function)  and S3 (implemented?) probably cover a lot of cases, I don't want to really support every method here as we can't support the long tail of the distribution. It should probably be the user's responsibility to provide a way to expose the remote data with network mounts, local copies, etc.

File: radarpipeline/io/ingestion.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+from radarpipeline.io.reader import Reader
+from radarpipeline.project.sparkengine import SparkEngine
+
+from typing import Dict
+
+logger = logging.getLogger(__name__)
+
+
+class CustomDataReader():
```
Comment (afolarin) [inline radarpipeline/io/ingestion.py:11]: What is the distinction here between ingestion and reader?

**Review by afolarin — APPROVED** [review]
LGTM

File: radarpipeline/io/downloader.py
Comment (Hsankesara) [reply radarpipeline/io/downloader.py:1]: Agreed, I think SFTP and S3 would cover the majority of cases. Anything else would need to be sorted by the user. I haven't implemented S3 yet but it's on my TODO list.

File: radarpipeline/io/ingestion.py (hunk 2 above)
Comment (Hsankesara) [reply radarpipeline/io/ingestion.py:11]: Nothing, I think ingestion is the filename. I tried to keep the function names which are exposed to the user simple and straight. That's why I named it that way.

**Pull request merged**